### DISCUSSION TITLE
Ls-Dyna Unrecognized option

### DISCUSSION THREAD

**Discussion by Serince** [OpenRadioss/OpenRadioss#488] [Ideas]
Starter shows  "Unrecognized option" error for the LS-Dyna k file unsupported cards and stops the conversion. However, if the problems are showing on the screen and rad file output is generated anyway, user can make changes for unsupported cards by editing rad file.

**Comment by PaulAltair** [comment]
This is not possible, the way the reader works, no rad starter file (_0000.rad) is created, the .k Input is  interpreted directly in order to generate restart files read by the engine. If a Dyna keyword option is unrecognised the only workaround is to modify that input.

**Reply by Serince** [reply]
Thank you for quick answer, Unfortunately it is not possible to modify unsupported card to make it supported cards. Such as, radioss supporting EOS_gruneisen or EOS linear_polynomial but If Ls-Dyna keyword file including this cards, starter is not mapping to Radioss. All helping documents in here is about usage and building of radioss. From last week, I try to understand code scheme. It is hard to enter directly in code. Is it possible to put some documentation about code scheme. I really interest with LS-Dyna mapping, but I didnt get the code logic.

**Reply by PaulAltair** [reply]
Yes sure, I appreciate that, Dyna Input support as it stands today is listed in the user documentation, see https://2022.help.altair.com/2022.1/hwsolvers/rad/topics/solvers/rad/ls_dyna_keywords_r.htm development of support of the format to date was focused on core keywords for structural analysis. So far as I am aware, EOS entries are not included yet, In the case the input includes cards that are unsupported then the model will not run.

You may think of the keywords listed in documentation as an alternate input format.

So if the input is not recognised the modification would be to remove it or replace it with some alternative that is available.

Obviously if the model requires EOS and that is not yet supported in the Keyword format, then this presents a problem.

To produce equivalent model behaviour currently, you would need to convert the whole model to block format first and use equivalent /EOS cards.

There is development in progress to allow mixing block and keyword input in a model which may help this type of application in future.

Or the alternative is to code support for the 'missing' keywords directly as OpenRadioss input. I'm not familiar with that process, I hope others may provide more assistance here.

**Reply by AlexanderKazantchev** [reply]
> So if the input is not recognised the modification would be to remove it or replace it with some alternative that is available.
> 
> Obviously if the model requires EOS and that is not yet supported in the Keyword format, then this presents a problem.
> 
> To produce equivalent model behaviour currently, you would need to convert the whole model to block format first and use equivalent /EOS cards.
> 
> There is development in progress to allow mixing block and keyword input in a model which may help this type of application in future.
> 
> Or the alternative is to code support for the 'missing' keywords directly as OpenRadioss input. I'm not familiar with that process, I hope others may provide more assistance here.

Is there a way to add some additional ls-dyna keywords? For example, I see that Starter returns errors on keywords *EOS and some plastic materials with anisotropy. May be somebody can say, is there a way to modify sources of the Starter for supply such keywords?

**Comment by MarkusSchulten** [comment]
This would be great, if Radioss could generate a _0000.rad file out of k-file for individual editing and applying SPH or ALE or AMS to basic model of LS-PrePost :).

**Reply by elequiniou** [reply]
Dear Markus,

Thanks a lot for your interest in this project and your constructive remark.
What is under development is to allow mix input as mentioned by Paul earlier. This will allow to include OpenRadioss options (an EOS in Block format for instance) not currently available or supported under the *Keyword format (LS-Dyna compatible input). In parallel we are working on opening an API to help the community expand this *Keyword format for OpenRadioss.
We are working hard to provide both options as soon as possible. So, stay tuned!

Best Regards,

Eric.
